src/Commands/NotificationQueuer.php

```diff
@@ -274,24 +274,50 @@ private function doListen($media_type_id, bool $once = FALSE): void {
         ->note(dt('Waiting for a notification from mpx after notification ID @id...', ['@id' => $notification_id]));
       $notifications = $this->listener->listen($media_source, $notification_id);
 
-      // We need a reference to the last notification so we can set the last
-      // notification ID even if all notifications are filtered out.
-      $last_notification = end($notifications);
-
-      // Check to see if there were no notifications and we got a sync response.
-      // @see https://docs.theplatform.com/help/wsf-subscribing-to-change-notifications#tp-toc10
-      if ($last_notification->isSyncResponse()) {
-        $this->logger()->info(dt('All notifications have been processed.'));
-        $more_to_consume = FALSE;
-      }
-      else {
-        $this->filterAndQueue($media_type, $notifications);
-        $more_to_consume = !$once;
-      }
+      $more_to_consume = $this->processNotifications($notifications, $media_type, $once);
+    }
+  }
+
+  /**
+   * Process received notifications.
+   *
+   * @param \Lullabot\Mpx\DataService\Notification[] $notifications
+   *   Array of notifications.
+   * @param \Drupal\media\MediaTypeInterface $media_type
+   *   Media type for which notifications are being processed.
+   * @param bool $once
+   *   (optional) Run once instead of until all notifications are processed.
+   *
+   * @return bool
+   *   TRUE if there are more notifications to consume, otherwise FALSE.
+   */
+  private function processNotifications(array $notifications, MediaTypeInterface $media_type, bool $once = FALSE): bool {
+    // We did not receive any notifications, connection may have timed out.
+    if (empty($notifications)) {
+      $this->logger()->info(dt('Connection to MPX has timed out.'));
+      $this->io()->note(dt('Connection to MPX has timed out.'));
```

Comment:
Seems odd to do this twice. If I use `--verbose` I'll see this message twice. Perhaps remove the log call entirely?

Reply:
Yeah, will do, thanks.